### PR TITLE
PCI 578 create basic spring boot app

### PR DESCRIPTION
**Pull request by rjadhav-ch**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	</dependencies>
+
+	<build>
+		<plugins>
+
+			<plugin>
+				<groupId>org.springframework.boot</groupId>
+				<artifactId>spring-boot-maven-plugin</artifactId>
+				<configuration>
+					<mainClass>${start-class}</mainClass>
```
Comment (jsimons1989) [inline pom.xml]: Where is ${start-class} defined?

File: routes.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+app_name: orders.api.ch.gov.uk
+group: api
+# XXX Make sure all services with /scudOrders/ based routes have a weight LOWER than 1000
```
Comment (jsimons1989) [inline routes.yaml]: Is this service going to have endpoints `/scudOrders/`? If not, can this comment be removed please?

File: src/main/java/uk/gov/companieshouse/orders/api/controller/HealthcheckController.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package uk.gov.companieshouse.orders.api.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Returns HTTP OK response to allow Mesos/Marathon to ensure a healthy service is running
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/orders/api/controller/HealthcheckController.java]: This JavaDoc shouldn't refer to `Mesos/Marathon`. The deployment could change which would make the comment invalid.

File: src/main/resources/banner.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+   ____           __                  ___    ____  ____
+  / __ \_________/ /__  __________   /   |  / __ \/  _/
+ / / / / ___/ __  / _ \/ ___/ ___/  / /| | / /_/ // /
+/ /_/ / /  / /_/ /  __/ /  (__  )  / ___ |/ ____// /
+\____/_/   \__,_/\___/_/  /____/  /_/  |_/_/   /___/
```
Comment (jsimons1989) [inline src/main/resources/banner.txt]: Can this not be committed to master please? It isn't referenced anywhere else in this PR.

File: src/test/java/uk/gov/companieshouse/orders/api/controller/HealthcheckControllerIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@AutoConfigureMockMvc
+@SpringBootTest
+class HealthcheckControllerIntegrationTest {
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    @DisplayName("Successfullly returns health status")
```
Comment (jsimons1989) [inline src/test/java/uk/gov/companieshouse/orders/api/controller/HealthcheckControllerIntegrationTest.java]: Typo in `Successfullly`

File: src/test/java/uk/gov/companieshouse/orders/api/controller/HealthcheckControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+@ExtendWith(MockitoExtension.class)
+public class HealthcheckControllerTest {
+
+    @InjectMocks
+    private HealthcheckController controllerUnderTest;
+
+    @Test
+    @DisplayName("Health check confirms health with HTTP 200")
+    public void applicationHealthcheckRunsSuccessfully(){
+        // Given the app is running
```
Comment (jsimons1989) [inline src/test/java/uk/gov/companieshouse/orders/api/controller/HealthcheckControllerTest.java]: These comments aren't needed

File: pom.xml (hunk 1 above)
Comment (rjadhav-ch) [reply pom.xml]: There's now a property for 'start-class'.

File: routes.yaml (hunk 2 above)
Comment (rjadhav-ch) [reply routes.yaml]: This has been removed.

File: src/main/java/uk/gov/companieshouse/orders/api/controller/HealthcheckController.java (hunk 3 above)
Comment (rjadhav-ch) [reply src/main/java/uk/gov/companieshouse/orders/api/controller/HealthcheckController.java]: I have updated the JavaDoc to remove references to Mesos/Marathon.

File: src/main/resources/banner.txt (hunk 4 above)
Comment (rjadhav-ch) [reply src/main/resources/banner.txt]: This has been removed.

File: src/test/java/uk/gov/companieshouse/orders/api/controller/HealthcheckControllerIntegrationTest.java (hunk 5 above)
Comment (rjadhav-ch) [reply src/test/java/uk/gov/companieshouse/orders/api/controller/HealthcheckControllerIntegrationTest.java]: Fixed.

File: src/test/java/uk/gov/companieshouse/orders/api/controller/HealthcheckControllerTest.java (hunk 6 above)
Comment (rjadhav-ch) [reply src/test/java/uk/gov/companieshouse/orders/api/controller/HealthcheckControllerTest.java]: The comment has been removed.

File: src/main/java/uk/gov/companieshouse/orders/api/OrdersApiApplication.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+@SpringBootApplication
+public class OrdersApiApplication {
+
+	public static final String APPLICATION_NAMESPACE = "orders.api.ch.gov.uk";
+
+	public static void main(String[] args) {
+		SpringApplication.run(OrdersApiApplication.class, args);
+	}
+
+}
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/orders/api/OrdersApiApplication.java]: As discussed, there are questions about this health check that I don't think we will be able to answer in this PR.
1. What is the expected contract? Will this depend on the Mesos version used?
2. Is the health check mechanism currently disabled for all apps in our AWS environments?
3. Assuming that a health check implementation is required, could it be achieved with less code duplication by using an actuator health check implementation for example?

**Comment by VaughanJackson123** [comment]
@jsimons1989 Good point. We have now raised spike https://companieshouse.atlassian.net/browse/PCI-584 to answer the health check questions.

**Pull request merged**